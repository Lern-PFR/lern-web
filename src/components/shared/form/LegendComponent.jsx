import styled from 'styled-components';
import {
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
} from 'styled-system';
import PropTypes from 'prop-types';
import { groupedLabelStyle } from 'theme/formStyles';

const StyledLegendComponent = styled('legend')(
	{
		...groupedLabelStyle,
	},
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
);

/**
 * @name LegendComponent
 * @description A component used to display styled label element.
 *
 * @author Christopher Walker
 *
 * @param {string}	children		: Test to display as the legend.
 */
const LegendComponent = ({ children, ...otherProps }) => (
	<StyledLegendComponent {...otherProps}>
		{children}
	</StyledLegendComponent>
);

LegendComponent.propTypes = {
	children: PropTypes.string.isRequired,
};

export default LegendComponent;
