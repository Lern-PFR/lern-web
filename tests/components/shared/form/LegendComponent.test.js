import { shallow } from 'enzyme';
import LegendComponent from 'components/shared/form/LegendComponent';
import { cleanup } from '@testing-library/react';

describe('LegendComponent', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match previous snapshot', () => {
		const sut = (<LegendComponent>Test legend</LegendComponent>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
