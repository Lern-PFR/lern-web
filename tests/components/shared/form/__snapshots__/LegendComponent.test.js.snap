// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendComponent should match previous snapshot 1`] = `
<styled.legend>
  Test legend
</styled.legend>
`;
